pages/blog/jhipster-jdl-to-asyncapi.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Convert JHipster JDL to AsyncAPI v2/v3 with ZenWave SDK
+date: 2023-06-17T06:00:00+01:00
+type: Engineering
+tags:
+  - ZenWave SDK
+  - Code Generator
+  - JHipster
+  - DSL
+cover: /img/posts/zenwave/jhipster-to-asyncapi-banner.png
+canonical: https://zenwave360.github.io/Domain-Driven-Design/JDL-Domain-Language/JHipster-As-IDL-for-AsyncAPIv2
+authors:
+  - name: Ivan Garcia Sainz-Aja
+    photo: /img/avatars/ivangsa.webp
+    link: https://www.linkedin.com/in/ivangarciasainzaja/
+    byline: Software Architect at Sngular. Speaker. Building ZenWave 360º
+excerpt: 'Because writting YAML by hand is now fun... You can generate AsyncAPI v2/v3 from  JHipster JDL models with ZenWave SDK.'
+---
+
+# Generating AsyncAPI definition files from JDL with ZenWaveSDK
+
+Writing YAML by hand is no fun, but you can simplify the process of writing AsyncAPI definition files by using a Domain Specific Language (DSL).
+
+[JHipster Domain Language (JDL)](https://www.jhipster.tech/jdl/intro) is a Domain Specific Language (DSL) used to define the domain model of a web application. With JDL, you can describe the entities, relationships, and constraints of your system in a concise and readable way.
+
+[Zenwave SDK](https://zenwave360.github.io/zenwave-sdk/) is set of tools to generate (and reverse engineering) code from JDL and API-First models like AsyncAPI and OpenAPI.
+
+Thanks to ZenWave SDK, you can convert JDL models into AsyncAPI definition files. This can save time and effort in the development process while ensuring that your APIs follow best practices and standards.
+
+## JDL Example
+
+```jdl
+@aggregate
+entity Customer {
+    username String required minlength(3) maxlength(250)
+    password String required minlength(3) maxlength(250)
+    email String required minlength(3) maxlength(250)
+    firstName String required minlength(3) maxlength(250)
+    lastName String required minlength(3) maxlength(250)
+}
+entity Address {
+    street String
+    city String
+    country String
+    zipCode String
+}
+
+relationship OneToOne {
+    Customer{address} to Address{customer}
+}
+```
+
+
+## Generating AsyncAPI definition files from JDL with ZenWaveSDK
+
+> See [JDL To AsyncAPI Generator](https://zenwave360.github.io/zenwave-sdk/plugins/jdl-to-asyncapi/) for a complete list of options and [GitHub repository](https://github.com/zenwave360/zenwave-sdk) for install instructions.
+
+Because JDL can only describe static aspects of your models and doesn't cover dynamic behaviour, ZenWave SDK can only infer CRUD operations from your entities, generating:
```

Comment:
```suggestion
Because JDL can only describe static aspects of your models and doesn't cover dynamic behavior, ZenWave SDK can only infer CRUD operations from your entities, generating:
```